From 753c7c7cc0d778f8e200d7cdc3def2fd76fdf1b9 Mon Sep 17 00:00:00 2001
From: Ondrej Kubik <ondrej.kubik@canonical.com>
Date: Tue, 12 Sep 2017 21:58:48 +0000
Subject: [PATCH 5/5] setting emmc system-boot partition

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>
---
 include/configs/dragonboard410c.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/dragonboard410c.h b/include/configs/dragonboard410c.h
index 6647f67..76b375d 100644
--- a/include/configs/dragonboard410c.h
+++ b/include/configs/dragonboard410c.h
@@ -72,8 +72,8 @@
 
 #define BOOT_TARGET_DEVICES(func) \
 	func(USB, usb, 0) \
-	func(MMC, mmc, 1) \
 	func(MMC, mmc, 0) \
+	func(MMC, mmc, 1) \
 	func(DHCP, dhcp, na)
 
 #include <config_distro_bootcmd.h>
@@ -123,7 +123,7 @@ REFLASH(dragonboard/u-boot.img, 8)\
 #define CONFIG_ENV_IS_IN_FAT
 #define CONFIG_FAT_WRITE
 #define FAT_ENV_INTERFACE      "mmc"
-#define FAT_ENV_DEVICE_AND_PART        "0:0"
+#define FAT_ENV_DEVICE_AND_PART        "0:8"
 #define FAT_ENV_FILE           "uboot.env"
 #define CONFIG_ENV_VARS_UBOOT_CONFIG
 #define CONFIG_SYS_REDUNDAND_ENVIRONMENT
-- 
2.7.4

