From 16c961cfb8f461cb697194979a38c90558057ccd Mon Sep 17 00:00:00 2001
From: Ondrej Kubik <ondrej.kubik@canonical.com>
Date: Tue, 5 Feb 2019 15:14:22 +0000
Subject: [PATCH] suport patch for dragonboard410c for ubuntu-core

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>
---
 include/configs/dragonboard410c.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/configs/dragonboard410c.h b/include/configs/dragonboard410c.h
index cf02108..cf81d84 100644
--- a/include/configs/dragonboard410c.h
+++ b/include/configs/dragonboard410c.h
@@ -89,9 +89,10 @@ REFLASH(dragonboard/u-boot.img, 8)\
 	"pxefile_addr_r=0x90100000\0"\
 	BOOTENV
 
-#define CONFIG_ENV_SIZE			0x2000
+#define CONFIG_ENV_SIZE			SZ_128K
 #define CONFIG_SYS_MMC_ENV_DEV		0	/* mmc0 = emmc, mmc1 = sd */
 #define CONFIG_SYS_MMC_ENV_PART 2 /* Set env partition to BOOT2 partition */
+#define CONFIG_SYS_REDUNDAND_ENVIRONMENT
 
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + SZ_8M)
-- 
2.7.4

