From d0804969c440ede42b863f98d249c83feca07564 Mon Sep 17 00:00:00 2001
From: Ondrej Kubik <ondrej.kubik@canonical.com>
Date: Tue, 12 Sep 2017 21:58:48 +0000
Subject: [PATCH 1/1] setting emmc system-boot partition

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>
---
 include/configs/dragonboard410c.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/dragonboard410c.h b/include/configs/dragonboard410c.h
index 6647f67..b37c1b6 100644
--- a/include/configs/dragonboard410c.h
+++ b/include/configs/dragonboard410c.h
@@ -123,7 +123,7 @@ REFLASH(dragonboard/u-boot.img, 8)\
 #define CONFIG_ENV_IS_IN_FAT
 #define CONFIG_FAT_WRITE
 #define FAT_ENV_INTERFACE      "mmc"
-#define FAT_ENV_DEVICE_AND_PART        "0:0"
+#define FAT_ENV_DEVICE_AND_PART        "0:9"
 #define FAT_ENV_FILE           "uboot.env"
 #define CONFIG_ENV_VARS_UBOOT_CONFIG
 #define CONFIG_SYS_REDUNDAND_ENVIRONMENT
-- 
2.7.4

