From 9b16682ee148ad4193d84fdda4c739f35b59290b Mon Sep 17 00:00:00 2001
From: Ondrej Kubik <ondrej.kubik@canonical.com>
Date: Wed, 21 Jun 2017 17:23:58 +0000
Subject: [PATCH 4/5] Rolling back change breaking dragonboard410c

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>
---
 drivers/mmc/sdhci.c | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/drivers/mmc/sdhci.c b/drivers/mmc/sdhci.c
index b745977..c94d58d 100644
--- a/drivers/mmc/sdhci.c
+++ b/drivers/mmc/sdhci.c
@@ -72,7 +72,6 @@ static int sdhci_transfer_data(struct sdhci_host *host, struct mmc_data *data,
 				unsigned int start_addr)
 {
 	unsigned int stat, rdy, mask, timeout, block = 0;
-	bool transfer_done = false;
 #ifdef CONFIG_MMC_SDHCI_SDMA
 	unsigned char ctrl;
 	ctrl = sdhci_readb(host, SDHCI_HOST_CONTROL);
@@ -90,23 +89,17 @@ static int sdhci_transfer_data(struct sdhci_host *host, struct mmc_data *data,
 			       __func__, stat);
 			return -EIO;
 		}
-		if (!transfer_done && (stat & rdy)) {
+		if (stat & rdy) {
 			if (!(sdhci_readl(host, SDHCI_PRESENT_STATE) & mask))
 				continue;
 			sdhci_writel(host, rdy, SDHCI_INT_STATUS);
 			sdhci_transfer_pio(host, data);
 			data->dest += data->blocksize;
-			if (++block >= data->blocks) {
-				/* Keep looping until the SDHCI_INT_DATA_END is
-				 * cleared, even if we finished sending all the
-				 * blocks.
-				 */
-				transfer_done = true;
-				continue;
-			}
+			if (++block >= data->blocks)
+				break;
 		}
 #ifdef CONFIG_MMC_SDHCI_SDMA
-		if (!transfer_done && (stat & SDHCI_INT_DMA_END)) {
+		if (stat & SDHCI_INT_DMA_END) {
 			sdhci_writel(host, SDHCI_INT_DMA_END, SDHCI_INT_STATUS);
 			start_addr &= ~(SDHCI_DEFAULT_BOUNDARY_SIZE - 1);
 			start_addr += SDHCI_DEFAULT_BOUNDARY_SIZE;
-- 
2.7.4

